include/onnxruntime/core/framework/tensor_shape.h

```diff
@@ -37,6 +37,7 @@ class TensorShape : private std::vector<int64_t> {
   TensorShape(const int64_t* dimension_sizes, size_t dimension_count);
 
   TensorShape(const std::vector<int64_t>& dims);
+  TensorShape(std::vector<int64_t>&& dims);
```

Comment:
It's not needed. std::vector<int64_t> can be cast into TensorShape  forcely.

Reply:
Please see the ReinterpretBaseType function in this file

Reply:
see comment above.